examples/thread_name_with_extension.json

```diff
@@ -0,0 +1,37 @@
+[
+  {"ph": "M", "pid": 44888, "tid": 476837, "name": "thread_name", "args": {"name": "476837 "}},
+  {"ph": "E", "pid": 44888, "tid": 476837, "ts": 2.000000},
+  {"ph": "B", "pid": 44888, "tid": 476837, "ts": 2.000000, "name": "started_tracing"},
+  {"ph": "M", "pid": 44888, "tid": 476842, "name": "thread_name", "args": {"name": "476842 "}},
+  {"ph": "E", "pid": 44888, "tid": 476842, "ts": 66.000000},
+  {"ph": "B", "pid": 44888, "tid": 476842, "ts": 66.000000, "name": "started"},
+  {"ph": "E", "pid": 44888, "tid": 476842, "ts": 69.000000},
+  {"ph": "B", "pid": 44888, "tid": 476842, "ts": 69.000000, "name": "wants_gvl"},
+  {"ph": "M", "pid": 44888, "tid": 476843, "name": "thread_name", "args": {"name": "476843 "}},
+  {"ph": "E", "pid": 44888, "tid": 476843, "ts": 82.000000},
+  {"ph": "B", "pid": 44888, "tid": 476843, "ts": 82.000000, "name": "started"},
+  {"ph": "E", "pid": 44888, "tid": 476843, "ts": 85.000000},
+  {"ph": "B", "pid": 44888, "tid": 476843, "ts": 85.000000, "name": "wants_gvl"},
+  {"ph": "M", "pid": 44888, "tid": 476844, "name": "thread_name", "args": {"name": "476844 "}},
+  {"ph": "E", "pid": 44888, "tid": 476844, "ts": 93.000000},
+  {"ph": "B", "pid": 44888, "tid": 476844, "ts": 93.000000, "name": "started"},
```

Comment:
Not anything this PR touches or changes, but it's interesting that we're not getting beyond microsecond precision on macOS, but we do so on Linux. At some point I need to perhaps look into the other options beyond `CLOCK_MONOTONIC` to see if we can make it tighter (I think there's a `_RAW` variant....)

Reply:
Interesting.